src/main/java/com/cloudbees/hudson/plugins/folder/AbstractFolderDescriptor.java

```diff
@@ -75,8 +100,74 @@ public String getCategoryId() {
         return r;
     }
 
+    /**
+     * Gets the {@link FolderIconDescriptor}s applicable for this folder type.
+     *
+     * @since FIXME
+     */
+    public List<FolderIconDescriptor> getIconDescriptors() {
+        List<FolderIconDescriptor> r = new ArrayList<FolderIconDescriptor>();
+        for (FolderIconDescriptor p : FolderIconDescriptor.all()) {
+            if (p.isApplicable(clazz.asSubclass(AbstractFolder.class))) {
+                r.add(p);
+            }
+        }
+        StaplerRequest request = Stapler.getCurrentRequest();
+        if (request != null) {
+            AbstractFolder folder = request.findAncestorObject(AbstractFolder.class);
+            if (folder != null) {
+                return DescriptorVisibilityFilter.apply(folder, r);
+            }
+        }
+        return r;
+    }
+
+    // TODO remove once baseline 2.0
+    public String getDescription() {
+        Stapler stapler = Stapler.getCurrent();
+        if (stapler != null) {
+            try {
+                WebApp webapp = WebApp.getCurrent();
+                MetaClass meta = webapp.getMetaClass(this);
+                Script s = meta.loadTearOff(JellyClassTearOff.class).findScript("newInstanceDetail");
```

Comment:
This is unnecessary. `AbstractFolderDescriptor` subclasses already define `newInstanceDetail.jelly` as

```xml
<j:jelly xmlns:j="jelly:core">
  <j:out value="${it.description}"/>
</j:jelly>
```

and implement a `String getDescription()` method, specifically so that we do not need such hacks.